src/Persistence/Search/Query/RangeFacetQueryBuilder.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Persistence\Search\Query;
+
+use Elastica\Query;
+use Elastica\Query\AbstractQuery;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\FacetConfig;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\PropertyConstraints;
+use Wikibase\DataModel\Services\Lookup\PropertyDataTypeLookup;
+
+class RangeFacetQueryBuilder implements FacetQueryBuilder {
+
+	public function __construct(
+		private readonly PropertyDataTypeLookup $dataTypeLookup
+	) {
+	}
+
+	public function buildQuery( FacetConfig $config, PropertyConstraints $constraints ): ?AbstractQuery {
+		$name = 'wbfs_' . $config->propertyId->getSerialization();
+
+		return match ( $this->dataTypeLookup->getDataTypeIdForProperty( $config->propertyId ) ) {
+			'quantity' => $this->buildQuantityQuery( $name, $constraints ),
+			'time' => $this->buildTimeQuery( $name, $constraints ),
+			default => null
+		};
+	}
+
+	private function buildQuantityQuery( string $name, PropertyConstraints $constraints ): AbstractQuery {
+		return new Query\Range(
```

Comment:
Same with above, should the return type be `Range` from  `Elastica\Query\Range` instead?